R/tab.R

```diff
@@ -1,73 +1,132 @@
 
 
-#' Tabulate data by intervals
-#'
-#' These functions call their related `chop_xxx` function, and call [table()]
-#' on the result.
-#'
-#' @param ... Passed to `chop`
-#'
-#' @return A [table()].
-#'
+#' @rdname chop
 #' @export
-#'
 #' @examples
 #' tab(1:10, c(2, 5, 8))
 #'
-#' tab_mean_sd(1:10)
-#'
-#' tab_equally(rpois(50, 4), groups = 4)
-tab <- function (...) {
-  default_table(chop(...))
+tab <-
+  function (
+    x,
+    breaks,
+    labels    = lbl_intervals(),
+    extend    = NULL,
+    left      = TRUE,
+    close_end = FALSE,
+    drop      = TRUE
+  ) {
+  default_table(
+    chop(
+      x         = x,
+      breaks    = breaks,
+      labels    = labels,
+      extend    = extend,
+      left      = left,
+      close_end = close_end,
+      drop      = drop
+    )
+  )
 }
 
 
-#' @rdname tab
+#' @rdname chop_width
 #' @export
-tab_width <- function (...) {
-  default_table(chop_width(...))
+#' @order 3
+#' @examples
+#' tab_width(1:10, 2, start = 0)
+#'
+tab_width <- function (x, width, start, ..., left = sign(width) > 0) {
+  default_table(
+    chop_width(
+      x     = x,
+      width = width,
+      start = start,
+      ...   = ...,
+      left  = left
+    )
+  )
 }
 
 
-#' @rdname tab
+#' @rdname chop_evenly
 #' @export
-tab_evenly <- function (...) {
-  default_table(chop_evenly(...))
+#' @order 3
+tab_evenly <- function (x, intervals, ..., close_end = TRUE) {
+    default_table(
+      chop_evenly(
+      x         = x,
+      intervals = intervals,
+      ...       = ...,
+      close_end = close_end
+    )
+  )
 }
 
 
-#' @rdname tab
+#' @rdname chop_n
 #' @export
-tab_n <- function (...) {
-  default_table(chop_n(...))
+#' @order 3
+#' @examples
+#' tab_n(1:10, 5)
+#'
+#' # fewer elements in one group
+#' tab_n(1:10, 4)
+#'
+tab_n <- function (x, n, ..., close_end = TRUE) {
+  default_table(chop_n(x = x, n = n, ... = ..., close_end = close_end))
 }
 
 
-#' @rdname tab
+#' @rdname chop_mean_sd
 #' @export
-tab_mean_sd <- function (...) {
-  default_table(chop_mean_sd(...))
+#' @order 3
+#' @examples
+#' tab_mean_sd(1:10)
+#'
+tab_mean_sd <- function (x, sd = 3, ...) {
+  default_table(chop_mean_sd(x = x, sd = sd, ... = ...))
 }
 
 
-#' @rdname tab
+#' @rdname chop_quantiles
 #' @export
-tab_quantiles <- function (...) {
-  default_table(chop_quantiles(...))
+#' @order 3
+#' @examples
+#' set.seed(42)
+#' tab_quantiles(rnorm(100), probs = 1:3/4, label = lbl_intervals(raw = TRUE))
+#'
+tab_quantiles <-
```

Comment:
Is there a reason for changing the signature of `tab_quantiles()` to be more explicit?

Reply:
Re more explicit signatures: I used the same arguments as the corresponding `chop_*` functions have. That way the user can see the main arguments and defaults for the `tab_*` function (as well). 

Also, only having `...` in the `tab_*` signature is ambiguous (now that they are in the same help files as `chop_*` and `brk_*` functions). For `chop_quantiles()` for example the \@param for `...` is 

> Passed to `chop()`, or for `brk_quantiles()` to `stats::quantile()`.

But this is not what `...` means for `tab_quantiles()`, because `...` would also entail `probs` for example, which is passed to `chop_quantiles()` first. 

Long story short: more explicit signatures seemed easier to explain and comprehend the function arguments.

> difference between `... = ...` and plain `...`

Ha, I wondered the same thing and honestly don't know. I wasn't aware that plain `...` was possible and only saw it after I already made some changes. I _think_ `... = ...` might be more reliable in edge cases, but I might be wrong. I looked shortly, but did not find anything definitive. I'll look further. 
Do you want me to change it back to plain `...` in the meantime?

Reply:
So long as it works, no big deal. I'll merge.